examples/amplify-react/src/App.js

```diff
@@ -0,0 +1,111 @@
+import React from 'react'
+import './App.css'
+import './loader.css'
+import ControlsForm from './ControlsForm'
+import VideoContainer from './VideoContainer'
+import { Auth } from 'aws-amplify'
+import Master from './master'
+import Viewer from './viewer'
+import { withAuthenticator, AmplifyGreetings } from '@aws-amplify/ui-react'
+
+class App extends React.Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      localViewSrcObject: '',
+      remoteViewSrcObject: '',
+      credentials: null
+    }
+
+    this.stopButtonHandler = this.stopButtonHandler.bind(this)
+    this.startMasterHandler = this.startMasterHandler.bind(this)
+    this.startViewerHandler = this.startViewerHandler.bind(this)
+
+    this.master = new Master(this.setState.bind(this), this.onStatsReport.bind(this), this.onRemoteDataMessage.bind(this))
+    this.viewer = new Viewer(this.setState.bind(this), this.onStatsReport.bind(this), this.onRemoteDataMessage.bind(this))
+  }
+
+  componentDidMount() {
+    window.addEventListener('beforeunload', this.stopButtonHandler)
+
+    window.addEventListener('unhandledrejection', e => {
+        console.error(e.reason.toString());
+        e.preventDefault();
+    })
+
+    window.addEventListener('error', e => {
+        console.error(e.message);
+        e.preventDefault();
```

Comment:
Is `e.preventDefault` to escape unhandled exceptions and compile-time errors when the page loads?

Reply:
Sample code incorporated from here: https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-js/blob/e6d3be4751d7afadd5e43780aee76c3fa9d096ab/examples/app.js#L94-L97

In react, I'd prefer not to use `window.addEventListener` at all, but this was added to maintain existing functionality